tests/unittests/config/test_cc_set_passwords.py

```diff
@@ -182,6 +182,105 @@ def test_failed_ssh_service_is_not_available(
         self.assertEqual(m_update_ssh_config.call_count, 0)
         self.assertEqual(m_subp.call_count, 0)
 
+    @mock.patch("cloudinit.distros.Distro.uses_systemd", return_value=False)
+    @mock.patch(MODPATH + "update_ssh_config", return_value=True)
+    @mock.patch("cloudinit.distros.subp.subp")
+    def test_failed_ssh_non_systemd_service_is_not_available_exit_code_25(
+        self, m_subp, m_update_ssh_config, m_uses_systemd
+    ):
+        """If the ssh service is not available in a system without systemd,
+        then no updates config, no restart and a generic message is logged.
+        """
+        cloud = self.tmp_cloud(distro="ubuntu")
+        cloud.distro.init_cmd = ["service"]
```

Comment:
The PR in 1449,  which only just landed while you were working on this PR, refactored some unittests to use pytest instead of subclassing from CiTestCase. #1449 provided a parameterized TestHandleSSHPwauth test case `test_no_restart_when_service_is_not_running`. Would these new tests make more sense if parameterized in either `TestHandleSSHPwauth.test_no_restart_when_service_is_not_running` or a new parameterized testcase instead of separate unittests?

Two other minor points:

- generally in unittests, I think we want to start avoiding calls to `self.tmp_cloud` and prefer `tests.unittests.util.get_cloud` instead as we try to retired tmp_cloud (and CiTestCase) where possible.
- I don't think we need to set `cloud.distro.init_cmd = ["service"]` on ubuntu distro because default is [`init_cmd 
 = ["service"]`](https://github.com/canonical/cloud-init/blob/main/cloudinit/distros/__init__.py#L78) and this only gets reset to "systemctl" if uses_systemd == True, which you've mocked in this test.

Reply:
Sorry for breaking the integration tests and thanks for pointing out the updates.

I have parametrized the tests.